### PR TITLE
Add support for MacOS & Windows wheels.

### PR DESCRIPTION
**Pull request by sbrodehl**
Hi there,

following up on #53 this PR adds support to build wheels for MacOS and Windows.

With the update, the following additional packages will be build and published:

-  pyhdf-0.10.4-cp310-cp310-macosx_10_9_x86_64.whl
-  pyhdf-0.10.4-cp36-cp36m-macosx_10_9_x86_64.whl
-  pyhdf-0.10.4-cp37-cp37m-macosx_10_9_x86_64.whl
-  pyhdf-0.10.4-cp38-cp38-macosx_10_9_x86_64.whl
-  pyhdf-0.10.4-cp39-cp39-macosx_10_9_x86_64.whl
-  pyhdf-0.10.4-pp37-pypy37_pp73-macosx_10_9_x86_64.whl
-  pyhdf-0.10.4-pp38-pypy38_pp73-macosx_10_9_x86_64.whl
-  pyhdf-0.10.4-cp310-cp310-win_amd64.whl
-  pyhdf-0.10.4-cp36-cp36m-win_amd64.whl
-  pyhdf-0.10.4-cp37-cp37m-win_amd64.whl
-  pyhdf-0.10.4-cp38-cp38-win_amd64.whl
-  pyhdf-0.10.4-cp39-cp39-win_amd64.whl
-  pyhdf-0.10.4-pp37-pypy37_pp73-win_amd64.whl
-  pyhdf-0.10.4-pp38-pypy38_pp73-win_amd64.whl

### CODE REVIEW

**Comment by sbrodehl** [comment]
When the Windows wheels are build with Github Actions, can the Appveyor pipeline then be deleted?

**Comment by sbrodehl** [comment]
PR #56 already adds tests for Windows :+1:

**Comment by sbrodehl** [comment]
I went ahead and removed the Travis and Appveyor pipelines and updated the readme.
One thing I was wondering: How are the anaconda packages build and deployed?

**Comment by fhs** [comment]
> How are the anaconda packages build and deployed?

https://github.com/conda-forge/pyhdf-feedstock
The conda-forge devs maintain it, not me.

**Comment by sbrodehl** [comment]
Good, so I didn't break anything there.
What else is missing in this PR to get this merged?

File: .github/workflows/package.yml [hunk 1]
```diff
@@ -18,49 +24,97 @@ jobs:
...
-        - name: purge old _linux_wheels
+            CIBW_ARCHS_LINUX: 'x86_64'  # restrict to 64bit builds
+            CIBW_ARCHS_WINDOWS: 'AMD64'  # restrict to 64bit builds
+            # (mac-os) Install hdf-4.2.15 from sources
+            CIBW_BEFORE_ALL_MACOS: >
+              brew install ninja &&
+              cd /tmp &&
+              git clone https://github.com/HDFGroup/hdf4.git &&
+              mkdir build && cd build &&
+              ../hdf4/configure --enable-shared --disable-static --disable-fortran --disable-netcdf --enable-java --enable-production --with-zlib --prefix=/usr/local &&
```
Comment (fhs) [inline .github/workflows/package.yml:55]: why do we need `--enable-java`?

**Review by fhs — COMMENTED** [review]
Don't we need to bump the version to 0.10.5 for a new release? The version is listed in `setup.py` and `doc/conf.py`.

File: .github/workflows/package.yml (hunk 1 above)
Comment (sbrodehl) [reply .github/workflows/package.yml]: We don't! I was lazy and just used the default configure parameters, sorry!

**Comment by sbrodehl** [comment]
I bumped the version in the last commit, that's it, I guess?

**Review by fhs — APPROVED** [review]
Thanks!

**Comment by sbrodehl** [comment]
It looks like the Appveyor build need to be disabled in the settings somewhere.
The build is still 'running', but without any project to build, see [pyhdf/builds/43476907](https://ci.appveyor.com/project/fhs/pyhdf/builds/43476907)

**Comment by fhs** [comment]
@sbrodehl Yep, I've removed Appveyor access to this repo. Hopefully that is enough.

**Pull request merged**